runtime/metricsview/executor_timestamps.go

```diff
@@ -0,0 +1,199 @@
+package metricsview
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/rilldata/rill/runtime/drivers"
+	"golang.org/x/sync/errgroup"
+)
+
+const (
+	defaultExecutionTimeout = time.Minute * 3
+)
+
+func (e *Executor) resolveDuckDBClickHouseAndPinot(ctx context.Context) (TimestampsResult, error) {
+	filter := e.security.RowFilter()
+	if filter != "" {
+		filter = fmt.Sprintf(" WHERE %s", filter)
+	}
+	timeDim := e.olap.Dialect().EscapeIdentifier(e.metricsView.TimeDimension)
+	escapedTableName := e.olap.Dialect().EscapeTable(e.metricsView.Database, e.metricsView.DatabaseSchema, e.metricsView.Table)
+
+	var watermarkExpr string
+	if e.metricsView.WatermarkExpression != "" {
+		watermarkExpr = e.metricsView.WatermarkExpression
+	} else {
+		watermarkExpr = fmt.Sprintf("MAX(%s)", timeDim)
+	}
+
+	rangeSQL := fmt.Sprintf(
+		"SELECT min(%[1]s) as \"min\", max(%[1]s) as \"max\", %[2]s as \"watermark\" FROM %[3]s %[4]s",
+		timeDim,
+		watermarkExpr,
+		escapedTableName,
+		filter,
+	)
+
+	rows, err := e.olap.Execute(ctx, &drivers.Statement{
+		Query:            rangeSQL,
+		Priority:         e.priority,
+		ExecutionTimeout: defaultExecutionTimeout,
+	})
+	if err != nil {
+		return TimestampsResult{}, err
+	}
+	defer rows.Close()
+
+	if rows.Next() {
+		var minTime, maxTime, watermark *time.Time
+		err = rows.Scan(&minTime, &maxTime, &watermark)
```

Comment:
This ends up sending `**time.Time` to `rows.Scan`, might not be well handled by all implementations. Also makes the pointers unsafe to access since they may still be nil. I think this would be safer:
```suggestion
		var minTime, maxTime, watermark time.Time
		err = rows.Scan(&minTime, &maxTime, &watermark)
```

Reply:
This is needed for duckdb. It returns `null` when there are no rows. Perhaps that is why we had different implementation for duckdb. We should probably go back to that. What do you think?

Reply:
And you checked it can't scan `null` into a `time.Time`? (Where the output would have `t.IsZero() == true`)

If yes, then no need for separate implementations, having an extra pointer layer is fine, as long as we check for `nil` before dereferencing.

Reply:
Ya it throws a type error. Saying it cant store nil into `time.Time`